code/modules/mob/mob_misc_procs.dm

```diff
@@ -807,3 +807,16 @@ GLOBAL_LIST_INIT(intents, list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM
 
 /mob/proc/attempt_listen_to_deadsay()
 
+
+/// Proc to PROPERLY set mob invisibility, huds gotta get set too!
+/mob/proc/set_invisible(invis_value)
+	if(invis_value)
+		invisibility = invis_value
+	else
```

Comment:
what if you want to set invis_value to 0? do you need a nullcheck as opposed to a not true check

Reply:
I think we'd want it to be initial in the vast majority of cases then, which is what we do in silicon chamo code